### PR TITLE
feat(Geo): add Geo class and amazonLocationServicesProvider - Geo milestone 1 - PR 3/6

### PR DESCRIPTION
**Pull request by TreTuna**
#### Description of changes

Geo milestone 1 - PR 3/6
Adds:
- initial `Geo` class
- initial `amazonLocationServicesProvider`
- configuration methods for both
- unit tests

#### Description of how you validated changes
Builds and unit tests all pass

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [x] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/geo/src/Geo.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+		this._config = Object.assign({}, this._config, amplifyConfig.Geo);
+
+		this._pluggables.forEach(pluggable => {
+			pluggable.configure(this._config);
+		});
+
+		if (this._pluggables.length === 0) {
+			const locationServicesProvider = new AmazonLocationServicesProvider();
+			locationServicesProvider.configure(this._config);
+			this.addPluggable(locationServicesProvider);
```
Comment (Amplifiyer) [inline packages/geo/src/Geo.ts:103]: Doesn't `addPluggable` calls the `configure` on provider as well with `this._config`?

File: packages/geo/src/types/Geo.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+export type GeoConfig = {
+	region?: string;
+	maps?: {
+		items: {};
+		default: string;
+	};
+	place_indexes?: {};
```
Comment (Amplifiyer) [inline packages/geo/src/types/Geo.ts:19]: This seems very provider specific. Any reasons to expose these types through category and not provider?

File: packages/geo/__tests__/Providers/AmazonLocationServicesProvider.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	});
+
+	describe('getProviderName', () => {
+		test('should return "AmazonLocationServices" when asked for Provider', () => {
+			const geo = new AmazonLocationServicesProvider();
+			expect(geo.getProviderName()).toBe('AmazonLocationServices');
+		});
+	});
+
+	describe('configure', () => {
```
Comment (Amplifiyer) [inline packages/geo/__tests__/Providers/AmazonLocationServicesProvider.test.ts:46]: We should add a test where the config is passed in the constructor of `AmazonLocationServicesProvider` for both variants of tests in this.

File: packages/geo/__tests__/Geo.test.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	});
+
+	describe('getModuleName', () => {
+		const geo = new GeoClass();
+		const moduleName = geo.getModuleName();
+
+		expect(moduleName).toBe('Geo');
+	});
+
+	describe('pluggables', () => {
```
Comment (Amplifiyer) [inline packages/geo/__tests__/Geo.test.ts:37]: A test for validating the default addition of `AmazonLocationServicesProvider` could be added.

**Review by Amplifiyer — COMMENTED** [review]
Just some small comments, mostly there.

File: packages/geo/__tests__/Providers/AmazonLocationServicesProvider.test.ts (hunk 3 above)
Comment (TreTuna) [reply packages/geo/__tests__/Providers/AmazonLocationServicesProvider.test.ts:46]: Sounds good

File: packages/geo/__tests__/Geo.test.ts (hunk 4 above)
Comment (TreTuna) [reply packages/geo/__tests__/Geo.test.ts:37]: Got it

File: packages/geo/src/Geo.ts (hunk 1 above)
Comment (TreTuna) [reply packages/geo/src/Geo.ts:103]: Good catch, I can remove configuring the provider here.

File: packages/geo/src/types/Geo.ts (hunk 2 above)
Comment (TreTuna) [reply packages/geo/src/types/Geo.ts:19]: Currently, the config coming from `aws-exports` is not specific to Amazon Location Services, thus these properties end up in the category configuration. We are discussing scoping them down and we can update this once that change is made in the CLI.

**Review by Amplifiyer — APPROVED** [review]
Looks good. The config can be updated in a separate PR

**Pull request merged**